### DISCUSSION TITLE
How to run TTT?

### DISCUSSION THREAD

**Discussion by 5angjun** [LearnLib/learnlib#124] [General]
Hi there.
i'm trying to execute TTT algorithm demo.
from https://github.com/LearnLib/learnlib-ttt

so i couldn't execute this TTT algorithm demo.
Can you show the example?

![image](https://github.com/LearnLib/learnlib/assets/76202024/1e9fd80f-e220-481f-bcdb-8db2c44659b0)

```

git clone https://github.com/LearnLib/learnlib-ttt.git
cd learnlib-ttt/experiments


sudo apt install default-jdk
sudo apt-get install maven
apt-get install gnuplot -y

./run.sh
./plot-results.sh
```
the ./run.sh result : there is no results folder after run ./run.sh
so ./plot-results.sh cannot find results.

**Comment by mtf90** [comment]
`run.sh` requires you to specify a profile which should be used. For example, `./run.sh pots2`.

However, note that this repository is unmaintained for several years now. For example, the code references old SNAPSHOT versions of LearnLib and the scripts may cause issues with current Python versions. There may be more troubles ahead.

**Reply by 5angjun** [reply]
Thanks for reply. So how to run TTT algorithm in latest version?

**Reply by mtf90** [reply]
If you are just interested in running the TTT learner (and not necessarily reproducing the benchmark results), I suggest you just use the latest LearnLib release. There exist [in-tree examples](https://github.com/LearnLib/learnlib/tree/learnlib-0.17.0/examples/src/main/java/de/learnlib/example) on how to set up the basic learning processes. For example, in `Example1.java` you can simply replace the `ClassicLStarDFA` learner with the `TTTLearnerDFA`, etc.

**Reply by 5angjun** [reply]
It would be great if there were more user-friendly instruction documentation for LearnLib. Thank you for the response, and I'll give it a try.

Thank you.

**Reply by mtf90** [reply]
Given that LearnLib is mainly a library you program against, I think having code-based examples is a fair compromise. You have a functional starting point which you can modify according to your individual needs and they are fairly easy to update during refactorings.

Do you need help with anything specific?

**Reply by 5angjun** [reply]
As with other open sources, I think simply providing general examples will be helpful to learnlib users. 

I don't need specific help right now, but if I get it, I'll post it on the discussion. thank you
